oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/TrackingCorruptIndexHandler.java

```diff
@@ -168,7 +168,13 @@ public class CorruptIndexInfo {
         private final String asyncName;
         private final String path;
         private final long lastIndexerCycleCount = indexerCycleCount;
-        private final Stopwatch watch = Stopwatch.createStarted(new ClockTicker(clock));
+        private final Stopwatch watch = Stopwatch.createStarted(new Ticker() {
+            @Override
+            public long read() {
+                return TimeUnit.MILLISECONDS.toNanos(clock.millis());
```

Comment:
I didn't look deeper, but shouldn't we use clock#getTime()? (might be overridden while #millis() isn't)

Reply:
`getTime()` does no exist in java.time.Clock, and I'd prefer new code to be generic.

Reply:
Aha. Yes, that may be a problem, so we should update "our" clock impl. That actually has a ticket already. Will work on it in parallel.